### DISCUSSION TITLE
API Preview: adding support for curl examples

### DISCUSSION THREAD

**Discussion by wbierbower** [pocketbase/pocketbase#5373] [Q&A] [answered]
Hi,  I love the API Preview section of PocketBase with the code examples for the JS and Dart SDKs.   Are there any plans to add a tab for bash curl commands?   Also, I was wondering how these SDK examples are generated in PocketBase incase I could contribute

**Comment by ganigeorgiev** [comment, marked as answer]
I guess it is doable but some of the example would be difficult to translate 1:1 with their SDK equivalent and I'm not sure how useful that would be.

The examples are static/manually typed and the code for them could be found in the [collections/docs Svelte components](https://github.com/pocketbase/pocketbase/tree/master/ui/src/components/collections/docs).

Note that with the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) and the upcoming v0.23 release (_there are no ETAs yet_) the API preview has been slightly updated in attempt to simplify a little the existing and new endpoints preview for the auth collections, so you may want to wait before spending time changing it. It looks like this at the moment (_the casing of the shown SDK methods is subject to change_):

![api_preview_screenshot](https://i.imgur.com/h1mJzx5.png)

**Reply by lucasRadio** [reply]
Hey @ganigeorgiev would you accept a PR if I add curl examples to the documentation?

**Reply by ganigeorgiev** [reply]
No sorry. It is already implemented in the new UI _(there are no ETAs when it will be released)_. It looks like this at the moment:

![curl_example_screenshot](https://i.imgur.com/VHTLeuT.png)

**Comment by whmeitzler** [comment]
This would be a genuinely nice quality-of-life enhancement. 
I actually did this as a weekend project and  it definitely helped make rapid dev easier.
Gani, if you want the code, you're welcome to it, but I suspect you;d prefer to just DIY. 
I did it in svelte, adding a  header with the bearer token of the curently signed-in admin.
You could even support realtime if you broke it into two parts -- a BASH script that exports the sub ID and subsequent calls to update the subscription.
